langchain/text_splitter.py

```diff
@@ -439,3 +439,292 @@ def __init__(self, **kwargs: Any):
             "",
         ]
         super().__init__(separators=separators, **kwargs)
+
+
+class GolangCodeTextSplitter(RecursiveCharacterTextSplitter):
```

Comment:
@ByronHsu @dev2049 

Instead of creating new classes for each language, could we parameterize by language? 

It'll save at least for lines of code per new language that's introduced, and also make it easier to see what languages are available.

```python

from typing import Union, Literal

Language = Union[Literal['go'], Literal['python'], Literal[...]]

class CodeTextSplitter(RecursiveCharacterTextSplitter):
    def __init__(self, language: Language, **kwargs: Any):
        separators = get_separators_for_language(language)
        super().__init__(separators=separators, **kwargs)
```

Reply:
Good idea! @dev2049 wdyt?

Reply:
im a fan! do we prefer typing language being a `Literal['python', 'go', ...]` or just making a string enum
```python
class Language(str, Enum):
  PYTHON = "python
  GO = "go"
  ...
```

Reply:
@dev2049 @eyurtsev Updated per our discussion!